### PR TITLE
7.12 Synthetics Docs Changes

### PR DESCRIPTION
**Pull request by andrewvc**
Updated for the 7.12.0 release. You'll want to test this by substituting `7.12.0-SNAPSHOT` as the version in any docker commands, since 8.0.0 isn't as relevant to actual users. When the release goes live the version will be the same, but without `SNAPSHOT`.

This includes significant changes to the config syntax.

### CODE REVIEW

File: docs/en/observability/synthetics-quickstart.asciidoc [hunk 1]
```diff
@@ -21,118 +23,96 @@ and a packaged todo application with a custom suite of tests.
...
-----
-<1> `heartbeat.docker.yml` is your Heartbeat configuration file.
-This is where you'll configure your synthetic test suites.
-<2> `run.sh` provides the main `docker run` command that pulls the
-Elastic synthetics image and runs Heartbeat.
 
 [discrete]
 [[synthetics-quickstart-step-two]]
-== Step 2: Update `heartbeat.docker.yml`
+== Step 2: Create a heartbeat.yml configuration file
```
Comment (paulb-elastic) [inline docs/en/observability/synthetics-quickstart.asciidoc:39]: Doesn't this need to be `heartbeat.docker.yml`?

File: docs/en/observability/synthetics-quickstart.asciidoc [hunk 2]
```diff
@@ -21,118 +23,96 @@ and a packaged todo application with a custom suite of tests.
...
 Let's take a quick look at each.
 
 Run an inline test::
 
-If you're running an inline, browser based test, you can use the traditional Heartbeat flow to completely
-configure your synthetic testing directly in `heartbeat.docker.yml`.
-An example is provided:
-+
+If you're running an inline browser based test, you can use the traditional Heartbeat flow to completely
+configure your synthetic testing directly in a file named `heartbeat.yml` as shown below.
```
Comment (paulb-elastic) [inline docs/en/observability/synthetics-quickstart.asciidoc:47]: Doesn't this need to be `heartbeat.docker.yml`?

File: docs/en/observability/synthetics-quickstart.asciidoc [hunk 3]
```diff
@@ -21,118 +23,96 @@ and a packaged todo application with a custom suite of tests.
...
-  script: |- <2>
-    step("load homepage", async () => {
-        await page.goto('https://www.elastic.co');
-    });
-    step("hover over products menu", async () => {
-        await page.hover('css=[data-nav-item=products]');
-    });
+  source:
+    inline:
+      script: |- <2>
```
Comment (paulb-elastic) [inline docs/en/observability/synthetics-quickstart.asciidoc:61]: Is the `|-` still needed?

File: docs/en/observability/synthetics-quickstart.asciidoc (hunk 1 above)
Comment (paulb-elastic) [inline docs/en/observability/synthetics-quickstart.asciidoc:39]: Doesn't this need to be `heartbeat.docker.yml`?

File: docs/en/observability/synthetics-quickstart.asciidoc [hunk 4]
```diff
@@ -21,118 +23,96 @@ and a packaged todo application with a custom suite of tests.
...
 ----
-heartbeat.synthetic_suites: <1>
 - name: Todos
-  path: "/opt/examples/todos" <2>
+  id: todos
+  type: browser
   schedule: "@every 1m"
+  source:
+    local:
+      path: "/opt/examples/todos" <1>
```
Comment (paulb-elastic) [inline docs/en/observability/synthetics-quickstart.asciidoc:89]: I don't think this path inside the container matches the path in the volume mount in the `docker run` below

File: docs/en/observability/synthetics-quickstart.asciidoc [hunk 5]
```diff
@@ -21,118 +23,96 @@ and a packaged todo application with a custom suite of tests.
...
 ----
-<1> Specify the name, path, and schedule of your test suite.
-<2> In this example, our library of synthetic tests live in the `/examples/todos` directory of `elastic/synthetics`.
+<1> In this example, our library of synthetic tests come from https://github.com/elastic/synthetics/tree/master/examples/todos[our todos example]. The path shown here corresponds to the path heartbeat will look at inside the docker container.
```
Comment (paulb-elastic) [inline docs/en/observability/synthetics-quickstart.asciidoc:91]: Might be worth calling out how this relates to the configuration, something along the lines:

> ...as defined with the `--volume` parameter to `docker run` as shown below

**Review by paulb-elastic — COMMENTED** [review]
Some inline comment suggestions added.

File: docs/en/observability/synthetics-quickstart.asciidoc (hunk 3 above)
Comment (andrewvc) [reply docs/en/observability/synthetics-quickstart.asciidoc:58]: Yes, that syntax lets you use multiline strings in yaml

File: docs/en/observability/synthetics-quickstart.asciidoc (hunk 2 above)
Comment (andrewvc) [reply docs/en/observability/synthetics-quickstart.asciidoc]: No, I've intentionally simplified everything to `heartbeat.yml`, we now make no reference to `run.sh` etc.

File: docs/en/observability/synthetics-quickstart.asciidoc (hunk 5 above)
Comment (andrewvc) [reply docs/en/observability/synthetics-quickstart.asciidoc]: Improved in the latest changes

File: docs/en/observability/synthetics-quickstart.asciidoc [hunk 6]
```diff
@@ -21,144 +23,111 @@ and a packaged todo application with a custom suite of tests.
...
-// Right now we do not publish docker images for patch releases
-// Docker pull should reference only the .0 bugfix of the minor
-// We can accomplish this with the following attribute `{minor-version}.0`
-
-[source,sh,subs="attributes"]
-----
-sh run.sh {minor-version}.0 \
-  '-E cloud.id=<cloud-id>' \
-  '-E cloud.auth=elastic:<cloud-pass>'
+  --volume="$PWD/heartbeat.yml:/usr/share/heartbeat/heartbeat.yml:ro" \
```
Comment (vigneshshanmugam) [inline docs/en/observability/synthetics-quickstart.asciidoc:102]: Would be good to mention that PWD corresponds to `todos` folder.

**Review by vigneshshanmugam — COMMENTED** [review]
We need to update the `create-test` docs with the new syntax as well - https://github.com/elastic/observability-docs/blob/de73e22328a71d36edfb8b1176dea87b64d246c4/docs/en/observability/synthetics-create-test.asciidoc#running-synthetic-tests

File: docs/en/observability/synthetics-quickstart.asciidoc (hunk 6 above)
Comment (vigneshshanmugam) [reply docs/en/observability/synthetics-quickstart.asciidoc:102]: Just saw the comment from the top, taking it back.

**Comment by vigneshshanmugam** [comment]
Tested it locally with 7.12.0-SNAPSHOT Image. Examples works properly for both inline and directory suites.

File: docs/en/observability/synthetics-quickstart.asciidoc [hunk 7]
```diff
@@ -21,144 +23,111 @@ and a packaged todo application with a custom suite of tests.
...
-[source,sh]
-----
-git clone https://github.com/elastic/synthetics.git &&\
-cd synthetics/examples/docker
-----
+== Step 1: Pull the latest synthetics docker image
 
-There are two files in `synthetics/examples/docker` that you'll need to edit:
+Elastic Synthetics is regularly updated during this experimental phase, even within the same version.
+You'll want to ensure you're running the latest docker image by running the below command.{synthetics-image}
```
Comment (bmorelli25) [inline docs/en/observability/synthetics-quickstart.asciidoc:29]: Was the use of the `{synthetics-image}` attribute accidental?

Screen Shot 2021-03-22 at 1 24 17 PM

File: docs/en/observability/synthetics-quickstart.asciidoc [hunk 8]
```diff
@@ -21,144 +23,111 @@ and a packaged todo application with a custom suite of tests.
...
+  --volume="$PWD:/opt/todos:ro" \
+  {synthetics-image} heartbeat -e \
+  -E output.elasticsearch.hosts=["localhost:9200"] \
+  -E output.elasticsearch.username=elastic \
+  -E output.elasticsearch.password=changeme
 ----
 
+Note the two `--volume` options which mount local directories into the container. The first mounts the `heartbeat.yml` file you've configured,
+into the expected location for `heartbeat.yml` for heartbeat, the second mounts the suite into the VM in the same location listed in the `source`
+section of our `heartbeat.yml`.
```
Comment (bmorelli25) [inline docs/en/observability/synthetics-quickstart.asciidoc:128]: I'm struggling to parse the long second sentence. What about something like this?

```suggestion
Note the two `--volume` options, which mount local directories into the container. The first mounts the `heartbeat.yml` file configured above,
into Heartbeat's expected location for `heartbeat.yml`. The second mounts the test suite into the VM in the same location listed in the `source`
section of our `heartbeat.yml`.
```

File: docs/en/observability/synthetics-quickstart.asciidoc (hunk 7 above)
Comment (andrewvc) [reply docs/en/observability/synthetics-quickstart.asciidoc:29]: No, I thought it'd be helpful to have a shortcut there.

Comment (andrewvc) [reply docs/en/observability/synthetics-quickstart.asciidoc:29]: Actually, now that I think of it, is there a variable we can just use for heartbeat?

Comment (bmorelli25) [reply docs/en/observability/synthetics-quickstart.asciidoc:29]: I'm still not quite sure what this accomplishes?

Screen Shot 2021-03-22 at 1 43 46 PM

Comment (bmorelli25) [reply docs/en/observability/synthetics-quickstart.asciidoc:29]: The usage in the code block makes sense

Comment (bmorelli25) [reply docs/en/observability/synthetics-quickstart.asciidoc:29]: > Actually, now that I think of it, is there a variable we can just use for heartbeat?

There's `{heartbeat}` for Heartbeat.

Comment (andrewvc) [reply docs/en/observability/synthetics-quickstart.asciidoc]: Switched to using that except for code where I think the literal value (which must be lowercase too) makes more sense

**Comment by andrewvc** [comment]
@vigneshshanmugam the latest push updates create test as well

**Review by vigneshshanmugam — APPROVED** [review]
Looks good.

File: docs/en/observability/synthetics-quickstart.asciidoc [hunk 9]
```diff
@@ -21,144 +23,111 @@ and a packaged todo application with a custom suite of tests.
...
-See <<synthetics-syntax>> for more information.
+<3> In this example, our library of synthetic tests come from https://github.com/elastic/synthetics/tree/master/examples/todos[our todos example]. The path shown here corresponds to the path {heartbeat} will look at inside the docker container.
+<4> Note that this path is local to the docker container, not the host system. You can have the todos files anywhere you like on the host, and map them to `/opt/todos` in the container.
 
 [discrete]
 [[synthetics-quickstart-step-three]]
-== Step 3: Run `run.sh`
+== Step 3: Run the container, connecting it to Elasticsearch
+
+Before we proceed, you'll need to retrieve your Elasticsearch credentials for either an https://www.elastic.co/guide/en/beats/heartbeat/current/configure-cloud-id.html[Elastic Cloud ID] or another https://www.elastic.co/guide/en/beats/heartbeat/current/elasticsearch-output.html[Elasticsearch Cluster].
```
Comment (bmorelli25) [inline docs/en/observability/synthetics-quickstart.asciidoc:84]: Ahh, sorry, there's a heartbeat ref for links too :)

```suggestion
Before we proceed, you'll need to retrieve your Elasticsearch credentials for either an {heartbeat-ref}/configure-cloud-id.html[Elastic Cloud ID] or another {heartbeat-ref}/elasticsearch-output.html[Elasticsearch Cluster].
```

**Review by bmorelli25 — APPROVED** [review]
Looks good!

**Pull request merged**